watertap/unit_models/anaerobic_digestor.py

```diff
@@ -0,0 +1,880 @@
+            rule=rule_material_balance,
+            doc="Unit level material balances",
+        )
+
+        def Sh2_conc_rule(self, t):
+            return self.liquid_phase.mass_transfer_term[t, "Liq", "S_h2"] == -1 * (
+                pyunits.convert(self.K_La, to_units=1 / pyunits.s)
+                * (
+                    self.liquid_phase.properties_out[t].conc_mass_comp["S_h2"]
+                    - 16
+                    * pyunits.kg
+                    / pyunits.kmol
+                    * pyunits.convert(
+                        self.KH_h2,
+                        to_units=pyunits.kmol / pyunits.m**3 * pyunits.Pa**-1,
+                    )
+                    * self.vapor_phase[t].p_sat["S_h2"]
+                )
+                * self.volume_liquid[t]
+            )
+
+        self.Sh2_conc = Constraint(
+            self.flowsheet().time,
+            rule=Sh2_conc_rule,
+            doc="Mass transfer rate of H2 gas vap",
+        )
+
+        def Sch4_conc_rule(self, t):
+            return self.liquid_phase.mass_transfer_term[t, "Liq", "S_ch4"] == -1 * (
+                pyunits.convert(self.K_La, to_units=1 / pyunits.s)
+                * (
+                    self.liquid_phase.properties_out[t].conc_mass_comp["S_ch4"]
+                    - 64
+                    * pyunits.kg
+                    / pyunits.kmol
+                    * pyunits.convert(
+                        self.KH_ch4,
+                        to_units=pyunits.kmol / pyunits.m**3 * pyunits.Pa**-1,
+                    )
+                    * self.vapor_phase[t].p_sat["S_ch4"]
+                )
+                * self.volume_liquid[t]
+            )
+
+        self.Sch4_conc = Constraint(
+            self.flowsheet().time,
+            rule=Sch4_conc_rule,
+            doc="Mass transfer rate of CH4 gas vap",
+        )
+
+        def Sco2_conc_rule(self, t):
+            return self.liquid_phase.mass_transfer_term[t, "Liq", "S_IC"] == -1 * (
+                pyunits.convert(self.K_La, to_units=1 / pyunits.s)
+                * (  # 0.0099 * pyunits.kmol / pyunits.m**3
+                    self.liquid_phase.reactions[t].conc_mol_co2
+                    - pyunits.convert(
+                        self.KH_co2,
+                        to_units=pyunits.kmol / pyunits.m**3 * pyunits.Pa**-1,
+                    )
+                    * self.vapor_phase[t].p_sat["S_co2"]
+                )
+                * self.volume_liquid[t]
+            ) * (1 * pyunits.kg / pyunits.kmole)
+
+        self.Sco2_conc = Constraint(
+            self.flowsheet().time,
+            rule=Sco2_conc_rule,
+            doc="Mass transfer rate of CO2 gas vap",
+        )
+
+        def flow_vol_vap_rule(self, t):
+            return self.vapor_phase[t].flow_vol == (
+                pyunits.convert(
+                    self.k_p, to_units=pyunits.m**3 / pyunits.s / pyunits.Pa
+                )
+                * (self.vapor_phase[t].pressure - 101325 * pyunits.Pa)
```

Comment:
Does the literature use gauge pressure for this constraint? Can probably add atmospheric pressure as a param to the adm1_properties_vapor model and use here. #909